### PR TITLE
fix: signin persistance across controllers

### PR DESCRIPTION
**Pull request by HunnySajid**
https://github.com/WebOfTrust/signify-browser-extension/issues/153

Signin structure in storage is updated.
Instead of array of signin object, it is now an object of arrays of signin objects.

E.g:

```
{
"agent1-ID":[signin1Obj,signin2Obj, signin3Obj],
"agent2-ID":[signin1Obj,signin2Obj]
}
```

### CODE REVIEW

File: src/pages/background/services/signify.ts [hunk 1]
```diff
@@ -150,6 +150,11 @@ const Signify = () => {
     return signed_headers;
   };
 
+  const getAgentID = (): string => {
```
Comment (2byrds) [inline src/pages/background/services/signify.ts:153]: Consider adding a call for getControllerID that accesses _client?.controller?.pre!

File: src/pages/background/services/signify.ts [hunk 2]
```diff
@@ -166,6 +171,7 @@ const Signify = () => {
     createAID,
     generatePasscode,
     bootAndConnect,
+    getAgentID,
```
Comment (2byrds) [inline src/pages/background/services/signify.ts:174]: add getControllerID

File: src/pages/background/resource/signin/index.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const getSigninsObject = async (): Promise<IObjectSignins> => {
+  const signinsObj = (await browserStorageService.getValue(
+    "signins"
+  )) as IObjectSignins;
+  return signinsObj ?? {};
+};
+
+export const getSignins = async (): Promise<ISignin[]> => {
+  const signinsObj = await getSigninsObject();
+  return signinsObj[signifyService.getAgentID()] ?? [];
```
Comment (2byrds) [inline src/pages/background/resource/signin/index.ts:23]: recommend using getControllerID()

File: src/pages/background/resource/signin/index.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+export const updateSignins = async (signins: ISignin[]) => {
+  const signinsObj = await getSigninsObject();
+  signinsObj[signifyService.getAgentID()] = signins;
```
Comment (2byrds) [inline src/pages/background/resource/signin/index.ts:28]: recommend using getControllerID()

**Review by 2byrds — CHANGES_REQUESTED** [review]

File: src/pages/background/services/signify.ts [hunk 5]
```diff
@@ -47,6 +48,8 @@ const Signify = () => {
       _client = new SignifyClient(agentUrl, passcode, Tier.low, bootUrl);
       await _client.boot();
       await _client.connect();
+      const state = await getState();
+      await userService.setControllerId(state?.controller?.state?.i);
```
Comment (HunnySajid) [inline src/pages/background/services/signify.ts:52]: @rodolfomiranda as client.state() is an async operation, it blocks the UI.
E.g if I try to fetch controllerId to attach with signin.
I have saved the controllerId into storage and fetch it from there.

**Review by 2byrds — APPROVED** [review]

**Pull request merged**